### DISCUSSION TITLE
issue: isDirty is updated to true only after entering the second digit

### DISCUSSION THREAD

**Discussion by PantherX99** [react-hook-form/react-hook-form#11013] [Q&A]
### Version Number

7.47.0

### Codesandbox/Expo snack

https://codesandbox.io/s/jolly-fast-74r8mf

### Steps to reproduce

1. Click "Open Dialog" and open Console
2. enter 1 digit into Menge input field and wait 1 second
3. even with delay, isDirty logs to false. 
4. enter a second digit, isDirty logs to true

Maybe I'm just blind and can't find my error but the behavior seems a bit odd to me.

### Expected behaviour

After the first digit is entered, isDirty should already be true.

In a larger component, the form also causes a re-render only after the first digit is entered. After that, it doesn't. It also causes a re-render when changing the date (not time) in one of the date pickers. Maybe this is related to that.

### What browsers are you seeing the problem on?

Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> Important: Make sure to provide all inputs' defaultValues at the useForm, so hook form can have a single source of truth to compare whether the form is dirty.

https://react-hook-form.com/docs/useform/formstate

**Reply by PantherX99** [reply]
Thanks, but have you tried it? The defaultValue was provided in the Controller. I changed it and set the defaultValues in the useForm hook but it still behaves the same.
The first keystroke is not recognized as isDirty.

**Reply by PantherX99** [reply]
> > Important: Make sure to provide all inputs' defaultValues at the useForm, so hook form can have a single source of truth to compare whether the form is dirty.
> 
> 
> 
> https://react-hook-form.com/docs/useform/formstate

@bluebill1049 it still does not work.

**Reply by PantherX99** [reply]
@bluebill1049 the error persists...

**Comment by proup24** [comment]
Possible solution:
https://github.com/orgs/react-hook-form/discussions/11049#discussioncomment-10654137
